src/main/java/parameter_service_demo/controller/ParameterController.java

```diff
@@ -98,4 +102,16 @@ public ParameterDto updateParameterById(@PathVariable Long id, @RequestBody @Val
     public void deleteParameterById(@PathVariable Long id) {
         parameterService.deleteById(id);
     }
+
+    @PostMapping("activate/{name}")
+    @Operation(summary = "Activates (locks) a parameter with ID provided in the path.")
+    @ApiResponse(responseCode = "200", description = "Parameter activated successfully.")
+    @ApiResponse(
+            responseCode = "400",
+            description = "Parameter is already activated.",
+            content = @Content(schema = @Schema(implementation = ErrorDto.class))
+    )
+    public List<ParameterDto> activateByName(@PathVariable String name, @RequestParam LocalDate activationDate) {
+        return parameterService.activateByName(name, activationDate);
+    }
```

Comment:
It's not stated that user can't activate activated parameters. Maybe share info of how many parameters with name were activated